src/image.rs

```diff
@@ -1,24 +1,254 @@
 //! Image module
 
-use winapi::um::wingdi::{GetDIBits, GetObjectW};
+use std::{
+    io::{self, Cursor, Write},
+    mem,
+    ops::{Deref, DerefMut},
+    os::raw::c_void,
+    ptr, slice,
+};
 
-use std::{ptr, slice, io, mem};
-use std::os::raw::{c_void, c_int, c_long, c_ulong};
-
-struct Dc(winapi::shared::windef::HDC);
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use winapi::{
+    shared::windef::HDC,
+    um::{
+        minwinbase::LPTR,
+        winbase::{LocalAlloc, LocalFree},
+        wingdi::{
+            CreateDIBitmap, GetDIBits, GetObjectW, BITMAP, BITMAPFILEHEADER, BITMAPINFO,
+            BITMAPINFOHEADER, BI_RGB, CBM_INIT, DIB_RGB_COLORS, RGBQUAD,
+        },
+        winuser::{EmptyClipboard, GetDC, ReleaseDC, SetClipboardData, CF_BITMAP},
+    },
+};
 
+struct Dc(HDC);
 impl Dc {
     fn new() -> Self {
-        Self(unsafe { winapi::um::winuser::GetDC(ptr::null_mut()) })
+        Self(unsafe { GetDC(ptr::null_mut()) })
     }
 }
-
 impl Drop for Dc {
     fn drop(&mut self) {
-        unsafe { winapi::um::winuser::ReleaseDC(ptr::null_mut(), self.0) };
+        unsafe { ReleaseDC(ptr::null_mut(), self.0) };
+    }
+}
+
+struct LocalMemory<T>(ptr::NonNull<T>);
+impl<T> LocalMemory<T> {
+    pub fn new(size: usize) -> Option<Self> {
+        let ptr = unsafe { LocalAlloc(LPTR, size) } as *mut _;
+        Some(Self(ptr::NonNull::new(ptr)?))
+    }
+
+    pub fn as_ptr(&mut self) -> *mut T {
+        self.0.as_ptr()
+    }
+}
+impl<T> Drop for LocalMemory<T> {
+    fn drop(&mut self) {
+        unsafe { LocalFree(self.0.as_ptr() as _) };
+    }
+}
+impl<T> Deref for LocalMemory<T> {
+    type Target = T;
+    fn deref(&self) -> &Self::Target {
+        unsafe { self.0.as_ref() }
     }
 }
+impl<T> DerefMut for LocalMemory<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { self.0.as_mut() }
+    }
+}
+
+///Bitmap image from clipboard
+pub struct Image {
+    ///Raw image data
+    pub bytes: Vec<u8>,
+}
+
+fn io_error(message: &'static str) -> io::Error {
+    io::Error::new(io::ErrorKind::Other, message)
+}
+
+impl Image {
+    pub(crate) fn from_handle(handle: ptr::NonNull<c_void>) -> io::Result<Self> {
+        let mut bitmap = BITMAP {
```

Comment:
`let mut bitmap: BITMAP = unsafe { mem::MaybeUninit::zeroed().assume_init() } `

Reply:
but why would we add unecessary unsafe there? why is this preferred instead of simply creating the struct?

Reply:
well it shortens code, but if you don't want to then it is fine :)